clang/lib/StaticAnalyzer/Checkers/StackAddrEscapeChecker.cpp

```diff
@@ -374,13 +395,13 @@ void StackAddrEscapeChecker::checkEndFunction(const ReturnStmt *RS,
 
     // Generate a report for this bug.
     const StringRef CommonSuffix =
-        "upon returning to the caller.  This will be a dangling reference";
+        " upon returning to the caller.  This will be a dangling reference";
```

Comment:
Is having two spaces after a dot a common convention we have? This is not a change in this PR, I'm just wondering if this was intentional or a typo.

Reply:
It is an older typographical convention that was historically common: https://en.wikipedia.org/wiki/Sentence_spacing

However, in the last 10-20 years it became obsolete almost everywhere, and I don't recall any other message where the static analyzer uses it. I'd guess that it was intentional (perhaps the personally preferred style of some contributor?), but we should probably return to single spacing.

Reply:
I found this style in a few other messages:

> Call to 'dispatch_once' uses the local variable 'once' for the predicate value.  Using such transient memory for the predicate is potentially dangerous.  Perhaps you intended to declare the variable as 'static'?

[link](https://github.com/llvm/llvm-project/blob/main/clang/test/Analysis/dispatch-once.m#L24)

> Object leaked: object allocated and stored into 'object' is returned from a function whose name ('CFGetRuleViolation') does not contain 'Copy' or 'Create'.  This violates the naming convention rules given in the Memory Management Guide for Core Foundation

[link](https://github.com/llvm/llvm-project/blob/main/clang/test/Analysis/retain-release-path-notes.m#L115C61-L115C316)

> The return value from the call to 'setuid' is not checked.  If an error occurs in 'setuid', the following code may execute with unexpected privileges

[link](https://github.com/llvm/llvm-project/blob/main/clang/test/Analysis/security-syntax-checks.m#L103)

> Function 'rand' is obsolete because it implements a poor random number generator.  Use 'arc4random' instead

[link](https://github.com/llvm/llvm-project/blob/main/clang/test/Analysis/security-syntax-checks.m#L138)

Overall, if lit-test messages stats is of any indication, it is pretty balanced:

    grep -R 'warning{.*\.  [A-Za-z]' clang/test/ | wc -l
    
-> 20 (double space after dot)
     
    grep -R 'warning{.*\. [A-Za-z]' clang/test/ | wc -l
    
-> 26 (single space after dot)